lib/net.c

```diff
@@ -78,6 +71,51 @@ static struct rte_eth_conf gatekeeper_port_conf = {
 	},
 };
 
+/*
+ * @src_port and @dst_port must be in big endian.
+ * By specifying the tuple (proto, src_port, dst_port),
+ * it can filter both IPv4 and IPv6 addresses.
+ */
+int
+ntuple_filter_add(uint8_t portid,
+	uint16_t src_port, uint16_t dst_port, uint16_t queue_id)
+{
+	int ret = 0;
+	struct rte_eth_ntuple_filter filter = {
```

Comment:
Can't we have a filter for IPv4 and another for IPv6? The reason being that the IPv4 filter can include the destination IP address, that is, the gatekeeper server's IPv4 address.

This is important because the IPv6 case will require more work: when GGU receives a IPv6 packet that is destined to the gatekeeper server, it will have to send that packet to the corresponding block to process it; that block could be a BP, RT, or GK.